### PR TITLE
JW7-1421 Update utils unit tests

### PR DESCRIPTION
**Pull request by pajong**

### CODE REVIEW

File: src/js/utils/stretching.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ define([
             if (value) {
                 value += ' ';
             }
-            value = 'translate(' + xoffset + 'px, ' + yoffset + 'px)';
+            value += 'translate(' + xoffset + 'px, ' + yoffset + 'px)';
```
Comment (donato) [inline src/js/utils/stretching.js]: This may be related : https://github.com/jwplayer/jwplayer/issues/778

File: test/unit/browser-test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    module('browser');
+
+    var testerGenerator = function (assert, method) {
+        return function (left, right, message) {
+            assert.strictEqual(method.apply(this, left), right, message);
+        };
+    };
+
+    test('browser.isInt', function(assert) {
```
Comment (donato) [inline test/unit/browser-test.js]: Surprise! Perhaps we should make isInt a private method within this utils file, since it doesn't seem related to browser checking.

File: test/unit/browser-test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        test([0.5],     false);
+        test(['10.'],    true);
+        test(['3.1'],   false);
+        test([NaN],     false);
+        test([null],    false);
+        test([undefined], false);
+    });
+
+    test('browser checks', function(assert) {
+        var fakeVersion = 'fakeVersion12';
```
Comment (donato) [inline test/unit/browser-test.js]: This could be a phantomjs only test, using fake user agents -- not sure if it's possible though.

File: test/unit/constants-test.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    'utils/constants'
+], function (constants) {
+    /* jshint qunit: true */
+
+    module('constants');
+
+    test('constants', function(assert) {
+        var skins = ['beelden', 'bekle', 'five', 'glow', 'roundster', 'six', 'stormtrooper', 'vapor'];
+
+        assert.equal(constants.repo, 'http://p.jwpcdn.com/player/v/');
```
Comment (donato) [inline test/unit/constants-test.js]: // What if we change the location?
assert.ok(_.isString(constants.repo))

File: test/unit/constants-test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+        assert.equal(constants.SkinsIncluded[0], 'seven');
+
+        for (var i = 0; i < constants.SkinsLoadable.length; i++) {
```
Comment (donato) [inline test/unit/constants-test.js]: array of strings?

File: test/unit/css-test.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        assert.ok(styleSheet.innerHTML.indexOf('test-selector{background-color:red}') >= 0, 'css correctly replaced');
+
+        css.clearCss('test-selector');
+
+        // check clearCss works correctly
+        assert.ok(styleSheet.innerHTML.indexOf('test-selector{background-color:red}') < 0, 'css correctly removed');
+    });
+
+    test('css.style', function(assert) {
+       var element = document.createElement('div');
```
Comment (donato) [inline test/unit/css-test.js]: Should we test camelCase as well?

File: test/unit/css-test.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+        css.style(element, styles);
+
+        assert.ok(element.getAttribute('style').indexOf('background-color: white') >= 0, 'css style background');
+        assert.ok(element.getAttribute('style').indexOf('z-index: 10') >= 0, 'css style z index');
+        assert.ok(element.getAttribute('style').indexOf('background-image: url(') >= 0, 'css style img');
+        assert.ok(element.getAttribute('style').indexOf('color: rgb(18, 52, 86)') >= 0, 'css style color');
+
+        // this should not do anything
+        css.style(null, styles);
```
Comment (donato) [inline test/unit/css-test.js]: Good idea, let's elaborate on "bad" test-cases

File: test/unit/dom-test.js [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+    module('dom');
+
+    test('dom.addClass', function (assert) {
+        var element = document.createElement('div');
+        assert.strictEqual(element.className, '', 'Created an element with no classes');
+
+        dom.addClass(element, 'class1');
+        assert.equal(element.className, 'class1', 'Added first class to element');
+
+        dom.addClass(element, 'class2');
```
Comment (donato) [inline test/unit/dom-test.js]: What if we add the same class many times? - it should only be there once

File: test/unit/dom-test.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
+define([
+    'utils/dom'
+], function (dom) {
+    /* jshint qunit: true */
+
+    module('dom');
+
+    test('dom.addClass', function (assert) {
```
Comment (donato) [inline test/unit/dom-test.js]: Also accepts an array of classes, or a string of space-delimited classes

File: test/unit/dom-test.js [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+    test('dom.styleDimension', function(assert) {
+        var percentage = dom.styleDimension('50%');
+        var px = dom.styleDimension('50');
+
+        // check style dimensions with percentage and px
+        assert.equal(percentage, '50%', 'percentage dimension test');
+        assert.equal(px, '50px', 'px dimension test');
+    });
+
+    test('dom.classList and dom.toggleClass', function(assert) {
```
Comment (donato) [inline test/unit/dom-test.js]: Separate these

File: test/unit/dom-test.js [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+        var elementA = document.createElement('div');
+        elementA.classList.add('class1');
+        elementA.classList.add('class2');
+
+        var elementB = document.createElement('div');
+        dom.addClass(elementB, ['a']);
```
Comment (donato) [inline test/unit/dom-test.js]: only test one thing at a time

File: test/unit/playerutils-test.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
+define([
+    'utils/playerutils',
+    'version'
+], function (playerutils, version) {
+    /* jshint qunit: true */
+
+    module('playerutils');
+
+    test('playerutils.versionCheck', function(assert) {
+        var versionCheck = playerutils.versionCheck('0 5');
```
Comment (donato) [inline test/unit/playerutils-test.js]: 0.5

File: test/unit/playerutils-test.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+        assert.ok(versionCheck, 'playerutils.versionCheck with valid version');
+
+        versionCheck = playerutils.versionCheck('10 1');
+        assert.notOk(versionCheck, 'playerutils.versionCheck with invalid version');
+
+        var jParts = version.split(/\W/);
```
Comment (donato) [inline test/unit/playerutils-test.js]: var tmp
jwplayer.version = '5.0'
playerUtils.versionCheck('5.1');
jwplayer.version = tmpVersion;

File: test/unit/simplemodel-test.js [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+    'utils/simplemodel'
+], function (simplemodel) {
+    /* jshint qunit: true */
+
+    module('simplemodel');
+
+
+    test('simplemodel', function(assert) {
+        var model = simplemodel;
+        assert.notOk(model.get('noExisting'), 'get with no attributes');
```
Comment (donato) [inline test/unit/simplemodel-test.js]: One day we should test the set/change events are propagated

File: test/unit/simplemodel-test.js [hunk 15]
```diff
@@ -0,0 +1,21 @@
+define([
+    'utils/simplemodel'
+], function (simplemodel) {
+    /* jshint qunit: true */
+
+    module('simplemodel');
+
+
+    test('simplemodel', function(assert) {
+        var model = simplemodel;
```
Comment (donato) [inline test/unit/simplemodel-test.js]: // Otherwise we are editing the model shared by everyone
model = _.extend({}, simpleModel);

File: test/unit/parser-test.js [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+
+    var testerGenerator = function (assert, method) {
+        return function (left, right, message) {
+            assert.strictEqual(method.apply(this, left), right, message);
+        };
+    };
+
+    test('parser.getAbsolutePath', function(assert) {
+        var path = parser.getAbsolutePath(null, null);
+        assert.notOk(path);
```
Comment (robwalch) [inline test/unit/parser-test.js]: path is false or an empty string? Explain with a description:

`assert.notOk(path, 'passing null values returns X');`

File: test/unit/parser-test.js [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+
+        path = parser.getAbsolutePath('https://testingUrl', null);
+        assert.equal(path, 'https://testingUrl');
+
+        path = parser.getAbsolutePath('path', 'base');
+        assert.ok(path.indexOf('path') >= 0);
+        assert.ok(path.indexOf('base') >= 0);
+
+        var test = testerGenerator(assert, parser.getAbsolutePath);
+        test(['.',   'https://example.com/alpha/beta/gamma'], 'https://example.com/alpha/beta');
```
Comment (robwalch) [inline test/unit/parser-test.js]: gamma looks like a directory, but is treated as a file. Maybe naming it that way in the test would make the test clearer.

File: test/unit/strings-test.js [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+    'utils/strings'
+], function (strings) {
+    /* jshint qunit: true */
+
+    module('strings');
+
+    test('strings.pad', function(assert) {
+        var str = strings.pad('test', 7, '1');
+
+        assert.equal(str, '111test', 'strings padding correctly done');
```
Comment (robwalch) [inline test/unit/strings-test.js]: Let's also test when the lend of the 1st param is >= the second param (no padding added)

File: test/unit/strings-test.js [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+    });
+
+    test('strings.extension', function(assert) {
+        var ext = strings.extension('invalid');
+        assert.notOk(ext, 'invalid path extension');
```
Comment (robwalch) [inline test/unit/strings-test.js]: Not sure I like the falsy tests. false or ""

File: test/unit/strings-test.js [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+        assert.equal(sec, 300, 'minutes input returns seconds');
+
+        sec = strings.seconds('1h');
+        assert.equal(sec, 3600, 'hours input returns seconds');
+
+        sec = strings.seconds('5');
+        assert.equal(sec, 5, 'string number input returns number');
+
+        sec = strings.seconds('1:01');
+        assert.equal(sec, 61, 'minute seconds input returns seconds');
```
Comment (robwalch) [inline test/unit/strings-test.js]: This parses everything up to HHH:MM:SS.sss right?

File: test/unit/css-test.js [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+
+    });
+
+    test('css.transform', function(assert) {
+        var element = document.createElement('div');
+
+        css.transform(element, 'test-val');
```
Comment (robwalch) [inline test/unit/css-test.js]: Invalid values fail in real browsers because they reject anything that doesn't comply to the spec: http://www.w3.org/TR/css3-transforms/#transform-functions

'none' or a valid function like 'scale(x,y)' or 'translate(x, y)' would work:
```
        css.transform(element, 'none');

        assert.equal(element.style.transform, 'none', 'css transform');
        assert.equal(element.style.MsTransform, 'none', 'css transform ms');
        assert.equal(element.style.MozTransform, 'none', 'css transform moz');
        assert.equal(element.style.OTransform, 'none', 'css transform o');
```

File: test/unit/scriptloader-test.js [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+
+    test('scriptloader with style', function(assert) {
+        // check style tag true creates stylesheet
+        var script = new scriptloader('styleTag', true);
+        var tag = script.makeTag('styleTag');
+        assert.ok(tag.href.indexOf('styleTag') >= 0, 'makeTag with isStyle true creates style tag');
+
+        // try to load the same script
+        var script2 = new scriptloader('styleTag', true);
+        script2.load();
```
Comment (jw-kaurand) [inline test/unit/scriptloader-test.js]: What are we testing with this second script?  No assert afterwards.

File: test/unit/stretching-test.js [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+        // parent width/height larger
+        var scale = stretching.stretch('none', element, 4, 2, 1, 1);
+        assert.ok(scale, 'none is always scaled');
+        assert.ok(element.style.cssText.indexOf('width: 1px; height: 1px') >= 0,
+            'element should be correctly scaled');
+
+        // parent width/height smaller
+        scale = stretching.stretch('none', element, 1, 1, 4, 2);
+        assert.ok(scale, 'none is always scaled');
+        assert.ok(element.style.cssText.indexOf('left: -1px; right: -1px; width: 4px; height: 2px') >= 0,
```
Comment (robwalch) [inline test/unit/stretching-test.js]: This test is failing in chrome. "element should be correctly scaled". Maybe remove the spaces or allow for less rigid matching of property values.

File: test/unit/playerutils-test.js (hunk 13 above)
Comment (pajong) [inline test/unit/playerutils-test.js]: @donato This does not work because playerutils is already initialized and use current version number.

File: test/unit/strings-test.js (hunk 20 above)
Comment (pajong) [inline test/unit/strings-test.js]: Yes it does - I will add test cases for that

**Comment by pajong** [comment]
this PR is dependent on: https://github.com/jwplayer/jwplayer/pull/794

**Comment by pajong** [comment]
test this please
